### ISSUE TITLE
Server build fails

### ISSUE THREAD

**Issue by dennwc** [bblfsh/server#29]
Server version: 54c71fc1e313430b3fda16e1e7b0509455cfd5fc
SDK version: https://github.com/bblfsh/sdk/commit/df3e0dadeb24ad49f38561f0955312f3024c82fe

Build log:

```
docker build -f Dockerfile.build -t bblfsh-server-build .
Sending build context to Docker daemon 112.9 MB
Step 1 : FROM golang:1.8-alpine
 ---> e7baf3b1a3a5
Step 2 : RUN apk add --no-cache git make musl-dev musl-utils gcc lvm2-dev btrfs-progs-dev
 ---> Using cache
 ---> fd6daebf7136
Step 3 : ENV GOPATH /go
 ---> Using cache
 ---> 48d0a86e3f57
Step 4 : WORKDIR /go/src/github.com/bblfsh/server
 ---> Using cache
 ---> 4f7d3d5252c2
Successfully built 4f7d3d5252c2
docker run --rm -v /home/dennwc/Go:/go bblfsh-server-build make build-internal
mkdir -p /go/src/github.com/bblfsh/server/build; \
for cmd in bblfsh; do \
        cd /go/src/github.com/bblfsh/server/cmd/${cmd}; \
        go build --ldflags '-X main.version=master -X main.build=06-15-2017_19_51_11' -o /go/src/github.com/bblfsh/server/build/${cmd} .; \
done;
# github.com/bblfsh/server/cmd/bblfsh
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/tmp/go-link-688671004/000001.o: In function `vsnprintf':
/usr/include/x86_64-linux-gnu/bits/stdio2.h:77: undefined reference to `__vsnprintf_chk'
/tmp/go-link-688671004/000001.o: In function `child_func':
/home/dennwc/Go/src/github.com/bblfsh/server/vendor/github.com/opencontainers/runc/libcontainer/nsenter/nsexec.c:212: undefined reference to `__longjmp_chk'
/tmp/go-link-688671004/000001.o: In function `fprintf':
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/tmp/go-link-688671004/000001.o:/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: more undefined references to `__fprintf_chk' follow
/tmp/go-link-688671004/000001.o: In function `snprintf':
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64: undefined reference to `__snprintf_chk'
/tmp/go-link-688671004/000001.o: In function `fprintf':
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: undefined reference to `__fprintf_chk'
/tmp/go-link-688671004/000001.o:/usr/include/x86_64-linux-gnu/bits/stdio2.h:97: more undefined references to `__fprintf_chk' follow
collect2: error: ld returned 1 exit status

make: *** [Makefile:118: build-internal] Error 2
Makefile:115: recipe for target 'build' failed
make: *** [build] Error 2
```

**Comment by abeaumont** [comment]
It looks like an Alpine's C library version issue, I've not been able to reproduce it yet.

**Comment by smola** [comment]
Right. The thing is that we mount the host GOPATH into the container, and this might cause this kind of problems. @dennwc which distro do you use?

**Comment by bzz** [comment]
Have just been hit by this one on Ubuntu 16.04.2 LTS.

**Comment by juanjux** [comment]
@bzz for the last part, about the testing docker image, you need to do the `docker run -it bblfshdev bash` with the `--privileged` argument. Also, you shouldn't need to `go get` the server since that's already done when the image builds.

Other than that, I can reproduce this problem with the bblfsh-dev docker image but not on my local system (with or without the updated SDK dependency). Applying your PR fixes perfectly the problem on the docker image so I think this can be closed as soon as github.com/bblfsh/server/pull/52 is merged.

**Comment by smola** [comment]
This might be fixed by using a dedicated GOPATH.
@dennwc @bzz could you try `GOPATH=/my-custom-new-and-empty-gopath make build`?

**Comment by bzz** [comment]
@smola meanwhile PTAL #52 - it seems that this does not happen with `CGO_ENABLED=0`

**Comment by bzz** [comment]
May be I've misunderstood something, but `GOPATH=/tmp/gopath make build` fails for me
 
```
go get -v _/root/go/src/github.com/bblfsh/server
package _/root/go/src/github.com/bblfsh/server: unrecognized import path "_/root/go/src/github.com/bblfsh/server" (import path does not begin with hostname)
Makefile:82: recipe for target '_/root/go/src/github.com/bblfsh/server' failed
make: *** [_/root/go/src/github.com/bblfsh/server] Error 1
```

**Comment by smola** [comment]
@bzz Right. My command was wrong, since the server directory should be contained in GOPATH.

**Comment by bzz** [comment]
as #52 is merged now, @abeaumont shall we close this as well?

**Comment by abeaumont** [comment]
This should be fixed by #52.
